DPGAnalysis/MuonTools/python/nano_mu_hlt_cff.py

```diff
@@ -0,0 +1,777 @@
+    src = cms.InputTag("l1tTkMuonsGmt"),
+    cut = cms.string(""),
+    name = cms.string("L1TkMu"),
+    doc = cms.string(""),
+    extension = cms.bool(False),
+    variables = cms.PSet(
+        pt = Var("phPt()", "float", doc = "Physics pt"),
+        eta = Var("phEta()", "float", doc = "#eta"),
+        phi = Var("phPhi()", "float", doc = "#phi (rad)"),
+        dXY = Var("phD0()", "float", doc = "dXY (cm)"),
+        dZ = Var("phZ0()", "float", doc = "dZ (cm)")
+    )
+)
+
+# L2 offline seeds
+l2SeedTable = cms.EDProducer(
+    "SimpleTrajectorySeedFlatTableProducer",
+    src = cms.InputTag("hltL2OfflineMuonSeeds"),
+    cut = cms.string(""),
+    name = cms.string("l2_seed_offline"),
+    doc = cms.string(""),
+    extension = cms.bool(False),
+    variables = cms.PSet(
+        pt = Var("startingState().pt()", "float", doc = "p_T (GeV)"),
+        nHits = Var("nHits()", "int16", doc = ""),
+        localX = Var(
+            "startingState().parameters().position().x()",
+            "float",
+            doc = "local x of the seed",
+        ),
+        localY = Var(
+            "startingState().parameters().position().y()",
+            "float",
+            doc = "local y of the seed",
+        )
+    )
+)
+
+# L2 seeds from L1Tk Muons
+l2SeedFromL1TkMuonTable = cms.EDProducer(
+    "SimpleL2MuonTrajectorySeedFlatTableProducer",
+    src = cms.InputTag("hltL2MuonSeedsFromL1TkMuon"),
+    cut = cms.string(""),
+    name = cms.string("phase2_l2_seed"),
+    doc = cms.string(""),
+    extension = cms.bool(False),
+    variables = cms.PSet(
+        pt = Var("startingState().pt()", "float", doc = "p_T (GeV)"),
+        nHits = Var("nHits()", "int16", doc = ""),
+        localX = Var(
+            "startingState().parameters().position().x()",
+            "float",
+            doc = "local x of the seed",
+        ),
+        localY = Var(
+            "startingState().parameters().position().y()",
+            "float",
+            doc = "local y of the seed",
+        )
+    )
+)
+
+# L2 standalone muons
+l2MuTable = cms.EDProducer(
+    "SimpleTriggerTrackFlatTableProducer",
+    src = cms.InputTag("hltL2MuonsFromL1TkMuon"),
+    cut = cms.string(""),
+    name = cms.string("l2_mu"),
+    doc = cms.string(""),
+    extension = cms.bool(False),
+    variables = cms.PSet(
+        pt = Var("pt()", "float", doc = "p_T (GeV)"),
+        eta = Var("eta()", "float", doc = "#eta"),
+        phi = Var("phi()", "float", doc = "#phi (rad)"),
+        dXY = Var("dxy()", "float", doc = "dXY (cm)"),
+        dZ = Var("dz()", "float", doc = "dZ (cm)"),
+        t0 = Var("t0()", "float", doc = "t0 (ns)"),
+        nPixelHits = Var("hitPattern().numberOfValidPixelHits()", "int16", doc = ""),
+        nTrkLays = Var("hitPattern().trackerLayersWithMeasurement()", "int16", doc = ""),
+        nMuHits = Var("hitPattern().numberOfValidMuonHits()", "int16", doc = "")
+    )
+)
+
+# L2 standalone muons updated at vertex
+l2MuTableVtx = cms.EDProducer(
```

Comment:
A `clone`- based approach (cloning from `l2MuTable`) should also work for all `SimpleTriggerTrackFlatTableProducer` objects. 

Overall, you are always preserving the same structure of variables.